proto/one/enterprise/data/dataset_definition_v2.proto

```diff
@@ -0,0 +1,72 @@
+syntax = "proto3";
+
+import "google/protobuf/wrappers.proto";
+
+option csharp_namespace ="ONE.Models.CSharp.Enterprise.Data";
+
+message EnterpriseDataSourceDefinition {
+    // DataSoruce name
+    string name = 1;
+
+    // Optional: A list of key value configuration options for the datasource data retrieval
+    map<string, google.protobuf.StringValue> options = 2;
+}
+
+message EnterpriseDataTableQueryParameter {
+    // Name of the datatable query parameter
+    string name = 1;
+
+    // Type of the datatable query parameter
+    string type = 2;
+
+    // Value of the datatable query parameter
+    string value = 3;
+}
+
+message EnterpriseDataTableQuery {
+    // Name of DataTable referenced DataSource
+    string dataSourceName = 1;
+
+    // Datatable query parameters
+    repeated EnterpriseDataTableQueryParameter queryParameters = 2;
+}
+
+message EnterpriseDataTableField {
+    // DataTable field name. E.g. StartDate
+    string name = 1;
+    // DataTable field type. E.g. System.String, Guid, System.Double
+    string typeName = 2;
+    // DataTable reference telementry twinRefId
+    string telemetryId = 3;
+    // DataTable refernced DataSource field name
+    string field = 4;
+}
+
+message EnterpriseDataTableDefinition {
+    //Name of the datatable
+    string name = 1;
+ 
+    // Defines DataTable query
+    EnterpriseDataTableQuery Query = 2;
+    
+    // Datatable fields
+    repeated EnterpriseDataTableField fields = 3;    
+}
+
+message EnterpriseDataSetDefinition {
+
+    //Name of dataset
+    string name = 1;
+
+    //Title for the Report
+    string description = 2;
+
+    //Version of dataset
+    string version = 3;
+
+    //list of DataSources required by the dataTables
+    repeated EnterpriseDataSourceDefinition datasources = 4;
```

Comment:
Is there a one to one relationship between DataTables and DataSources or could each DataTable have multiple DataSources?  Having the two independent fields like this makes it seem like they are not related but I think they are correct?  Could you use a map<DataTableDefinition, repeated DataSourceDefinition>?  Then you would have a dictionary of the DataTables and the DataSources used to build that table.  Maybe I am misunderstanding the relationship between DataTables and DataSources here.

Reply:
Having worked with Steve I know that the JSON file you are working off of is almost as fluid as this code :)
So the dataTable definition would indicate which DataSourceDefinition to use and it would pull the definition from this list.
Unless a DataSourceDefinition can be used by multiple DataTables it seems like we would not need them to be separate lists and the dataSourceDefinition could be a child of the DataTable.
Maybe the better place for this conversation would be on the confluence page but whatever.

Reply:
I agree. A better place to describe the schema is in a wikipage with a class diagram. Let me create one class/type diagram and shared with you.